Async Execution (project scheduling)

somebody1234 opened the pull request:
### Pull Request Description
:warning: This is currently blocked waiting for the corresponding backend functionality.

- Frontend part of https://github.com/enso-org/cloud-v2/issues/1392
  - New modal to schedule project executions
  - New tab in asset right menu that lists project executions
  - Choose repeat interval (hourly, daily, weekly, monthly)
  - Choose parallel mode to control behavior when multiple executions are scheduled at the same time (ignore new execution if old one is not finished, cancel old execution, run both executions at the same time)
  - Select and display maximum execution time
  - "Use Current Version" button
  - Paused ("disabled") schedules

### Important Notes
None

### Screenshots

"New Schedule" dialog
![image](https://github.com/user-attachments/assets/a22d76ad-fd16-4a47-a5f5-faa845c0b2c4)

List of execution schedules:
![image](https://github.com/user-attachments/assets/8e2b126f-208e-46f0-ab65-cf6fd4bdd50b)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

somebody1234 committed:
Initial backend endpoint for listing project executions

somebody1234 committed:
Add rest of backend endpoints for managing executions

somebody1234 committed:
Initial UI for "new execution" modal

somebody1234 committed:
Basic display for schedules

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Switch `createProjectExecution` to `backendMutationOptions`

somebody1234 committed:
Fixes

somebody1234 committed:
Improve UI of "Executions" tab in Asset Panel

somebody1234 committed:
More adjustments for Project Execution side panel

somebody1234 committed:
Add button to delete project executions

somebody1234 committed:
More fine adjustments

somebody1234 committed:
More improvements to Project Executions side panel display

somebody1234 committed:
Sanitize irrelevant fields from Project Execution form

somebody1234 committed:
More fixes

somebody1234 committed:
Fix default values for forms

somebody1234 committed:
WIP: Switch to date picker

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Initial implementation of Date Picker component

somebody1234 committed:
Refactor form types

somebody1234 committed:
Adjustments to date picker and "new schedule" form

somebody1234 committed:
Add colors to execution settings to make them easier to distinguish

somebody1234 committed:
Fix size of DatePicker calendar popover

somebody1234 committed:
Temporarily log backend calls

somebody1234 committed:
Switch cancel button for "New Schedule" dialog from "cancel" variant to "outline" variant

somebody1234 committed:
Support maximum duration in minutes of an execution schedule

somebody1234 committed:
Add new API endpoints

somebody1234 committed:
UI for Project Execution enabled/disabled state and updating to latest version

somebody1234 committed:
Add text to Asset Panel when queries are loading

MrFlashAccount commented on app/dashboard/src/components/AriaComponents/Form/Form.tsx:
```diff
@@ -80,6 +78,17 @@ export const Form = React.forwardRef(function Form<
 
+  // Expose default values to controlled inputs like `Selector` and `MultiSelector`
+  const initialDefaultValues = React.useRef(defaultValues)
+  React.useEffect(() => {
```
This shall be moved to the `useForm` hook instead.
Also, I guess we might want to reset only the unchanged fields?

somebody1234 replied:
tried changing `form.reset(defaults)` to `form.reset(defaults, { keepDirtyValues: true })` but that's causing the timeout in minutes value (60) to not appear for some reason. maybe we're not managing the dirty state properly

MrFlashAccount commented on app/dashboard/src/components/AriaComponents/Form/components/Field.tsx:
```diff
@@ -20,8 +20,11 @@ import * as formContext from './useFormContext'
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  readonly form?: types.FormInstance<any, any, any>
+  readonly form?: Omit<types.FormInstance<any>, 'handleSubmit'> & {
+    readonly handleSubmit?: (...args: never) => unknown
```
Sorry but, Why?

somebody1234 replied:
i guess the new `types.FormInstance<any>` is not equivalent to the old `types.FormInstance<any, any, any>` and so the type of `handleSubmit` becomes something that errors somewhere else

somebody1234 replied:
```
Type 'FormInstance<Schema>' is not assignable to type 'FormInstance<any>'.
  Types of property 'handleSubmit' are incompatible.
    Type 'UseFormHandleSubmit<FieldValues<Schema>, TransformedValues<Schema>>' is not assignable to type 'UseFormHandleSubmit<any, any>'.
      Types of parameters 'onValid' and 'onValid' are incompatible.
        Type 'SubmitHandler<any>' is not assignable to type 'TransformedValues<Schema> extends undefined ? SubmitHandler<FieldValues<Schema>> : TransformedValues<Schema> extends FieldValues ? SubmitHandler<...> : never'.
```

MrFlashAccount replied:
![CleanShot 2024-08-20 at 11 06 35@2x](https://github.com/user-attachments/assets/9883aaa0-f46a-46d6-87a0-6a90c0d7f098)

Well, making the `handleSubmit` compatible is painful: https://github.com/enso-org/enso/blob/597dd633432ff2cab493aa8f2d3744d89e4e5cc6/app/dashboard/src/components/AriaComponents/Form/Form.tsx#L110

=)

somebody1234 replied:
broken link D:

somebody1234 replied:
ok so i've switched `Field.tsx` to use generics and it seems (?) to work?

somebody1234 committed:
Fix type error in `Field.tsx`

somebody1234 committed:
Move initial defaults to `useForm` hook

somebody1234 committed:
Keep dirty values when resetting the form to add initial default values

somebody1234 committed:
Minor fixes

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 commented:
marking as draft as it is still blocked by backend endpoints

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Switch Asset Panel buttons to use Tab Bar

somebody1234 committed:
Fix errors in `NewProjectExecutionModal`

somebody1234 committed:
Switch backend API for schedules back to arrays

somebody1234 committed:
Fix types for form `onSubmit`

somebody1234 committed:
Fix "new schedule" form not closing on submit

somebody1234 committed:
Type-safe inputs

somebody1234 committed:
Fix type errors in `ProjectExecution`

somebody1234 committed:
Fully migrate to type-safe inputs

somebody1234 committed:
Fix "Cancel" button on "new schedule" form

somebody1234 committed:
Restore multi-select form

somebody1234 committed:
Adjust text

somebody1234 commented:
@indiv0 note that if (when?) you do end up testing this PR that this PR mocks out the endpoints, the actual impls need to be uncommented (and the mocks removed) to test with an actual backend

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Animated resizing for dialogs

somebody1234 committed:
Fix framer-motion type error

somebody1234 committed:
Move "advanced mode" switch for "new schedule" dialog to top right

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Fix merge conflict

somebody1234 committed:
Fix lint errors

somebody1234 committed:
WIP: Fix errors

somebody1234 committed:
Add back Asset Panel tab for executions

somebody1234 committed:
Fix alignment of project execution tab; add invalidations to `backendHooks.tsx`

somebody1234 committed:
Remove colors from project executions

jdunkerley committed:
Fixes a few bits... (#11581)

- Minor fixes from past PR.
- Always use Table viz for Table.
- Fix display to `Always`  for any AWS required parameters.
- Add widget for `Redshift` credentials and alter `Username_And_Password` to require arguments.
- Fix display in `Data` module methods for required parameters.
- Fix `Statistic.bulk_widget`.
- Alter `Google_Analytics` so credentials the first parameter.
- Add support for storing the credential file in Enso cloud in `Google_Analytics`.

![image](https://github.com/user-attachments/assets/f34c4231-9f9f-468b-90e9-bbc7f2374d22)

Commit:
--jvm tries to find Java executable system-wide. (#11500)

Fixes `--jvm` option, given to the native image. This was failing on my machine, because when given `--jvm` option, the runner was trying to find the `java` executable from the distribution manager's runtime (on my system located in `~/.local/share/enso/runtime`) and it used the first runtime found. But the first runtime on my system is JDK 17.

The `--jvm` option now tries to:
- Find a JDK from the distribution manager that has the same version as the JDK used for building the engine.
- If there is not an exact version match, it tries to find a runtime from distribution manager that is *newer*.
- If none, fallback to system-wide search
- System-wide search tries to find `java` from `$JAVA_HOME` and from `$PATH`. But this is just a fallback.

# Important Notes
- Added test to Engine CI jobs that pass `--jvm` argument to a native image of engine-runner
- ea3af5ffbca6dc439690142251426b4c167bca98
- `runtime-version-manager` sbt project migrated to a JPMS module
- `engine-runner` now depends on `runtime-version-manager`.
- Removed unnecessary stuff in `runtime-version-manager` dealing with outdated `gu` Graal Updater utility.
- Extracted [GraalVersionManager](https://github.com/enso-org/enso/blob/1455b025cb8e5141409ea96b75577e8884265a70/lib/scala/runtime-version-manager/src/main/java/org/enso/runtimeversionmanager/components/GraalVersionManager.java) from [RuntimeVersionManager](https://github.com/enso-org/enso/blob/d2e8994700fd36228f7873f040668013e643190a/lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/components/RuntimeVersionManager.scala)

somebody1234 committed:
Make project executions full width

somebody1234 committed:
Move `AssetPanel` tabs into `AssetPanel/`

somebody1234 committed:
Named imports

somebody1234 committed:
Switch project execution options back to vertical

somebody1234 committed:
Better project execution date display

somebody1234 committed:
Show time buttons when hovering over project execution

somebody1234 committed:
Fix day display in project execution

somebody1234 committed:
Add ordinals for dates

somebody1234 committed:
Add project executions calendar tab

vitvakatu committed:
Disable EditContext API (#11554)

Fixes #11239

https://github.com/user-attachments/assets/eb57fb34-11f4-489b-9849-c93c7aa5e870

farmaazon committed:
Pasting image in Documentation Editor (#11547)

Fixes #10059

https://github.com/user-attachments/assets/a528e26a-b388-4a2a-9bf4-3ccc734373f6

# Important Notes
* I put the logic for project's files management to a single composable "projectFiles"

JaroslavTulach committed:
Multi value `Complex` test and robustness refactoring (#11525)

While working on #11482 and enhancing the tests suite with more tests based on `type Complex` a [getRootNode() did not terminate in 100000 iterations](https://github.com/enso-org/enso/pull/11525#issuecomment-2476171597) problem was discovered. Detailed investigation revealed that the existing `ReadArgumentCheckNode` infrastructure was able to create a **cycle** of parent pointers in the Truffle AST.

The problem was in intricate manipulation of the AST while rewriting internals in `ReadArgumentCheckNode`. This PR avoids such manipulation by _refactoring the type checking code_. `ReadArgumentNode` knows nothing about types anymore. When a type check is needed, `IrToTruffle` adds additional `TypeCheckValueNode.wrap` around the `ReadArgumentNode` - that breaks the **vicious circle**.

All the _type checks_ nodes are moved to its own package. All but one of the classes are made package private. The external API for doing _type checking_ is concentrated into `TypeCheckValueNode`.

somebody1234 committed:
Bump Node version in Nix flake (#11580)

- Bump Node.JS version to match `.node-version`:

https://github.com/enso-org/enso/blob/6b810ee1e6d92af889d14f5b547d8f517404a719/.node-version#L1

JaroslavTulach committed:
Allow (42 : Text & Integer) at the end of function (#11588)

Bypassing failing check.

JaroslavTulach committed:
HostValueToEnsoNode is not a builtin (#11593)

Bypassing failing check.

somebody1234 committed:
Add Calendar icon for execution calendar

somebody1234 committed:
Initial UI for calendar

vitvakatu committed:
Edges always end with arrows (#11590)

Bypassing the failing test.

vitvakatu committed:
New node colors (#11578)

* Show halo only on hover

* Fix numeric input widget. (#11561)

* Invert node colors, make edited widgets more prominent

---------

Co-authored-by: Paweł Grabarz <frizi09@gmail.com>

Skipping failing test in GUI packaging.
Nothing changes in Microsoft space.

kazcw committed:
Render tables in documentation. (#11564)

* Render tables in documentation.

Also:
- Separate parser for our flavor of Markdown from the CodeMirror integration;
  move the parser into ydoc-shared and use for Markdown line-wrapping.
- Introduce our own version of yCollab extension; initially just the upstream
  version translated to Typescript and our code style.
- Refactor CodeEditor.

* CHANGELOG, prettier

* Apply @farmaazon review.

* Fix

* Lint

* Cleanup

* Integration tests for GraphNodeComment

Also a little refactoring in preparation for new implementation.

* Workaround stuck CI

* Revert "Workaround stuck CI"

This reverts commit 74313842baa15d12c1a8be0529a16aa8ea204d83.

* Fix merge

---------

Co-authored-by: mergify[bot] <37929162+mergify[bot]@users.noreply.github.com>

AdRiley committed:
Complete migration to Dialect/Feature flags + SQLServer support for Datetime (#11527)

* Supports_Sort_Digits_As_Numbers

* case_insensitive_ordering

* Case_Insensitive_Ordering

* Case_Insensitive_Ordering

* Enable tests

* Case_Insensitive_Non_Ascii

* Order_By_Unicode_Normalization_By_Default

* Allows_Mixed_Type_Comparisons

* Supports_Unicode_Normalization

* NaN_Non_Comparable

* Distinct_Returns_First_Row_From_Group_If_Ordered

* Datetime support

* refactor

* refactor

* Refactor

* Move remaining flags

* Cleanup

* sbt javafmtAll

* Fix

* Fix tests

* Code review changes

* More date_diff month tests

* Fix sowflake

* Fix snowflake tests

Bypassing failing test.

somebody1234 committed:
WIP: Compute dates for project executions

somebody1234 committed:
Fix Datalink inputs (#11376)

* Stop validating optional Data Link fields

* Fix incorrect default value in Datalink input

* Stop `Autocomplete` fields from opening automatically without
`autoFocus`

* Increase E2E test timeout

* Fix E2E test race condition?

* Fix error message for empty string input

* Highlight active Datalink input

* Show description when Datalink input is invalid

* Fix Datalink input unfocusing when errors appear

* Fix `enso://` path error text

* Fix hover display of autocomplete items

* Move `Autocomplete` tooltip above Dropdown container

* Update Enso path validation

* Update test file and Enso File datalink regex

---------

Co-authored-by: mergify[bot] <37929162+mergify[bot]@users.noreply.github.com>
Bypassing failing tests.

farmaazon committed:
Ignore E2E CI tests on Windows (#11602)

The integration tests are constantly failing on Windows. For the time of investigation, they should be suppressed.

Commit:
Implement hasLanguage interop message for all enso objects (#11538)

* EnsoObject is an abstract class, not an interface.

- Also, EnsoObject exports hasLanguage and getLanguage interop messages.
- BranchRecord converted to class

* Implement public getters in BranchResult

* Fix compilation of EnsoFile

* Add test that all enso values must have language

* Revert EnsoException - remove

* DataflowError and PanicException implement hasLanguage and getLanguage

* DataflowError is not EnsoObject - change signatures in some builtins

* Add more members to Module.isMemberInvocable.

Keep in sync with doInvoke.

* Revert "DataflowError and PanicException implement hasLanguage and getLanguage"

This reverts commit b30f3961b724fb942ec1202e38175929e0baedeb.

* Update the test - test only non-primitive and non-exception values

* Fix indexes in CodeLocationsTest

* Add more members to Function.isMemberInvocable

Keep in sync with doInvoke.

* EnsoObject.toDisplayString delegates to toString method

* EnsoObject.toDisplayString is behind TruffleBoundary

* Warning exports InteropLibrary which delegates to value.

With the exception of toDisplayString message.

* WithWarnings needs to explicitly export toDisplayString.

It is not automatically delegated because it is implemented in the super type.

* EnsoObject.toDisplayString just throws AssertionError

* AssertionError is behind TruffleBoundary

* Implement toDisplayString on some truffle objects

* Warning exports WarningsLibrary

* Revert "Warning exports WarningsLibrary"

This reverts commit a06c672db5f5842871d0950fefb0b714dbb84727.

* Add some warnings test

* Warning.isNull is always false

Even if it wraps Nothing

* Add some unnecessary methods to fix the compilation

* EnsoObject.toDisplayString is abstract

* ImportExportScope.toDisplayString is behind TruffleBoundary.

This fixes native-image build of engine-runner.

* Hide some toDisplayString methods behind TruffleBoundary

This fixes native-image build of engine-runner.
Bypassing failing test.

somebody1234 committed:
Fix `firstProjectExecutionOnOrAfter`

somebody1234 committed:
WIP: Begin displaying project execution entries

JaroslavTulach committed:
Give `DebugBreakpointNode` some section to prevent NPE (#11594)

Fixes `NullPointerException` by giving `DebugBreakpointNode` a source section.

somebody1234 committed:
Use proxied auth endpoint for Cognito (#11387)

- Close https://github.com/enso-org/cloud-v2/issues/1552
- Switch to proxied endpoint for Cognito to enforce account lockout rate limits

# Important Notes
None

somebody1234 committed:
Wrap asset creation in a Promise (#11396)

- Close https://github.com/enso-org/cloud-v2/issues/1560
- Switch from "event"s to async functions so that new asset creation can be awaited

None

JaroslavTulach committed:
Invoke Warning.getValue().to_text and use it from `ReplDebuggerInstrument` (#11591)

Fixes #11569 by using `.to_text` on the right _warning's value_ (as introduced by #10842) and sharing the code with the instrument.

AdRiley committed:
Enable Union feature and make tests pass (#11587)

somebody1234 committed:
Fix keyboard shortcuts settings display (#11467)

- Add border to keyboard shortcuts to make them more distinct from each other
- Move scrollbar to table so the header becomes sticky

# Important Notes
None

somebody1234 committed:
Fix dialog entry animations and size (#11565)

- Stop resizing when `Dialog`s first render
- Fix `Dialog type="fullscreen"`s being smaller than before
- Check the "Discover" button in the top left to see a fullscreen modal.

# Important Notes
None

somebody1234 committed:
Make sure "new label" button is always visible even when labels panel has scrollbar (#11586)

- Move "new label" button outside of scroll container so that it is always visible

# Important Notes
None

marthasharkey committed:
fix filter node drilldown for number columns  (#11572)

GregoryTravis committed:
Don't log stack trace in cache tests (#11610)

somebody1234 committed:
"Save" and "Cancel" buttons for settings sections (#11290)

somebody1234 committed:
Fix project execution calendar display

somebody1234 committed:
Show list of project executions in calendar

somebody1234 committed:
List project executions for current month

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Add "every hour" text for hourly scheduled projects

somebody1234 committed:
Hide day in project executions on calendar

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Fix lint errors

MrFlashAccount commented:
Eventhough it's blocked by the backend, nothing's stopping us to merge it behind a flag

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Clean up `AssetPanel` components further

somebody1234 committed:
Sort executions by time

somebody1234 committed:
Adjust docs

somebody1234 committed:
Show "first occurrence field" in "hourly" repeat interval in simple view for
"New Schedule"

somebody1234 committed:
Also show "first occurrence" input in "new project execution" modal

somebody1234 committed:
Add "end date" field to "new project execution" modal; change granularity to "minute"

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
WIP: Fix schedules showing on previous day due to time zone issues

somebody1234 committed:
Change label back to "Maximum duration (minutes)" in "new schedule" modal

somebody1234 committed:
Equal widths for calendar columns

somebody1234 committed:
Add local storage state for local time zone

somebody1234 committed:
Factor in timezone offset when sending multi-selected schedule dates to backend

somebody1234 committed:
Prevent project icon from shrinking

somebody1234 committed:
Fix schedules not being listed for last day of month

somebody1234 committed:
WIP: Add start date and end date to "new schedule" backend endpoint wrapper

somebody1234 committed:
Fix errors

somebody1234 committed:
Fix timezone issue; show correct date in list view in calendar

somebody1234 committed:
Prevent schedules scheduled for midnight from showing up on two days

somebody1234 committed:
Highlight today's date in project schedule calendar

somebody1234 committed:
Show multiple times in `ProjectExecution`

somebody1234 committed:
Use minute of selected date for hourly new schedule dialog

somebody1234 committed:
Make `date` take priority over `projectExecution` for `ProjectExecution` display

PabloBuchu commented:
Lets simplify things and inspire with google calendar scheduler (without custom):
<img width="794" alt="Screenshot 2024-11-29 at 12 54 45" src="https://github.com/user-attachments/assets/e49d8f59-5669-4e3e-b6a3-7293a0ea4f55">
<img width="768" alt="Screenshot 2024-11-29 at 12 54 37" src="https://github.com/user-attachments/assets/6f70246b-60b1-4d88-8280-6711884353f7">

![image](https://github.com/user-attachments/assets/0ba349cc-f658-41a3-88bf-410168d14570)


Description:
1. Header: Schedule task
2. First row: Start date with date, time, timezone selectors 
3. 2nd row: repetition selector: Does not repeat (default), Hourly, Daily, 30th in Month, Last Sunday in Month (last 2 based on selected date)
4. Hourly: select first / last hour (range)
5. Daily: Select which days of the week should run 
6. 30th in Month, Last Sunday in Month: Choose which months,

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Initial update to new API

somebody1234 committed:
Fix `Selector`

somebody1234 committed:
Fix display of project execution repeats

somebody1234 committed:
WIP: Add tests for `nextProjectExecutionDate` and `firstProjectExecutionOnOrAfter`

somebody1234 committed:
Fix project execution tests

somebody1234 committed:
Fix display of project schedule repeat times

somebody1234 committed:
Fix repeat times for project executions with monthly repeats

somebody1234 committed:
Fix yearly repeats for project schedules; UI for date/week and day schedule repeats

somebody1234 committed:
Fix `AnimatedBackground` layout positioning being outdated

somebody1234 committed:
Adjust JSDoc comment formatting

somebody1234 committed:
Fix text

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Simplify new execution modal

somebody1234 committed:
WIP: Simplify UI

somebody1234 committed:
Display correct week number in "new schedule" modal

somebody1234 committed:
Filter out "last weekday" repeat type if selected date is not on last week

somebody1234 committed:
Switch repeat type to dropdown; add react-hook-form dropdown

somebody1234 committed:
Rearrange input fields in project schedule modal

somebody1234 committed:
Inline "new schedule" form into calendar; highlight days on which new schedule is triggered

somebody1234 committed:
Add "none" repeat type for project schedules

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Fix error...

somebody1234 committed:
Add boolean flag to disable mocked endpoints

somebody1234 committed:
Fix errors

somebody1234 committed:
Fix errors

somebody1234 committed:
Fix errors

somebody1234 committed:
Merge branch 'develop' into wip/sb/async-execution

somebody1234 committed:
Remove mocks for async execution

somebody1234 committed:
Disable executions and fix text for local backend

somebody1234 committed:
Fix new execution form defaults and off by one error for days

somebody1234 committed:
Remove code from wrong directory

somebody1234 committed:
Fix `DatePicker` calendar not appearing

somebody1234 committed:
Prevent form default submit action for project executions calendar

somebody1234 committed:
Add invalidations for project execution mutations

somebody1234 committed:
Fix errors

somebody1234 committed:
Revert "Remove mocks for async execution"

This reverts commit 95abb558c5a7cc15d5c3b25a96584e0150e3bcca.

somebody1234 committed:
Remove `mode="wait"` in `AnimatedBackground` because there are no longer any layouts that are animating when the parent is also animating

somebody1234 committed:
Hide "parallel mode" and "maximum duration" options behind feature flag

somebody1234 committed:
Fix class names

somebody1234 committed:
Fix canceling "new schedule" form in calendar

somebody1234 committed:
Fix infinite re-render loop

somebody1234 committed:
Reapply "Remove mocks for async execution"

This reverts commit 18cb1df8b8c09af459ef8e7277fc399c11d7969f.

somebody1234 committed:
Prettier

somebody1234 committed:
Fix errors

PabloBuchu commented:
1. First occurrence input display date time but I can only accurately edit date. It does not allow my to adjust hour or minutes. 
2. `Disabled (click to enable)` click resulted in 404 POST to `
https://xw0g8j3tsb.execute-api.eu-west-1.amazonaws.com/executions/undefined`
2.1 Same with delete
2.2 Same with sync
3. When are the executions fetched? I saw the request only once per project.. I removed them from db and still see them in GUI
4. Thats not much visible.. lets have a number of scheduled executions per day here
<img width="516" alt="Screenshot 2025-01-20 at 15 10 27" src="https://github.com/user-attachments/assets/455eab3f-a04a-4f1e-ac13-ff344e6da6f1" />
5. New schedule in executions tab opens a dialog while the same button in calendar opens bellow... We should have this consistent and always open in dialog
6. Also when I try to submit new schedule from the calendar tab I ends with Enso being reloaded

MrFlashAccount committed:
Devtools improvements

MrFlashAccount commented:
> When are the executions fetched? I saw the request only once per project.. I removed them from db and still see them in GUI

They're cached for a couple of minutes, as any query. guess we need refetch interval here aswell?

MrFlashAccount committed:
Fix critical bugs

MrFlashAccount committed:
Fixes

MrFlashAccount committed:
Fix devtools

MrFlashAccount committed:
Merge branch 'develop' into wip/sb/async-execution

# Conflicts:
#	app/gui/src/dashboard/App.tsx
#	app/gui/src/dashboard/components/AriaComponents/Checkbox/Checkbox.tsx
#	app/gui/src/dashboard/components/AriaComponents/Inputs/ComboBox/ComboBox.tsx
#	app/gui/src/dashboard/components/AriaComponents/Inputs/DatePicker/DatePicker.tsx
#	app/gui/src/dashboard/components/AriaComponents/Inputs/Input/Input.tsx
#	app/gui/src/dashboard/components/AriaComponents/Inputs/MultiSelector/MultiSelector.tsx
#	app/gui/src/dashboard/components/AriaComponents/Inputs/OTPInput/OTPInput.tsx
#	app/gui/src/dashboard/components/AriaComponents/Inputs/Selector/Selector.tsx
#	app/gui/src/dashboard/components/AriaComponents/Radio/RadioGroup.tsx
#	app/gui/src/dashboard/components/AriaComponents/Switch/Switch.tsx
#	app/gui/src/dashboard/components/AriaComponents/Text/Text.tsx

MrFlashAccount committed:
Fix lint issues

MrFlashAccount committed:
Fix ts

MrFlashAccount committed:
lint fuck it
